File: Questão3.java
class ContaBancaria {
    private String titular;
    private double saldo; 

    public ContaBancaria(String titular, double saldo) {
        this.titular = titular;
        this.saldo = saldo;
    }

    public void depositar(double valor) {
        if (valor < 0) {
            throw new IllegalArgumentException("Valor de depósito deve ser positivo.");
        }
        saldo += valor;
    }

    public void sacar(double valor) {
        if (valor < 0) {
            throw new IllegalArgumentException("Valor de saque deve ser positivo.");
        }
        if (valor > saldo) {
            throw new IllegalArgumentException("Saldo insuficiente para saque.");
        }
        saldo -= valor;
    }

    public double getSaldo() {
        return saldo;
    }
}

File: Questão3.go
package main

import (
    "errors"
    "fmt"
)

.
type ContaBancaria struct {
    Titular string
    saldo   float64 
}


func (c *ContaBancaria) Depositar(valor float64) error {
    if valor < 0 {
        return errors.New("valor de depósito deve ser positivo")
    }
    c.saldo += valor
    return nil
}


func (c *ContaBancaria) Sacar(valor float64) error {
    if valor < 0 {
        return errors.New("valor de saque deve ser positivo")
    }
    if valor > c.saldo {
        return errors.New("saldo insuficiente para saque")
    }
    c.saldo -= valor
    return nil
}


func (c *ContaBancaria) GetSaldo() float64 {
    return c.saldo
}
